velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -2186,5 +2187,28 @@ TEST_F(MinMaxByNTest, incrementalWindow) {
   }
 }
 
+TEST_F(MinMaxByNTest, peakMemory) {
+  auto data = makeRowVector({
+      // Each array has 10 elements.
+      makeArrayVector<int64_t>(
+          1'000, [](auto row) { return 100; }, [](auto row) { return row; }),
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row; }),
+  });
+
+  core::PlanNodeId aggId;
+  auto plan = PlanBuilder()
+                  .values({data})
+                  .singleAggregation({}, {"max_by(c0, c1, 2)"})
+                  .capturePlanNodeId(aggId)
+                  .planNode();
+
+  std::shared_ptr<exec::Task> task;
+  auto results = AssertQueryBuilder(plan).copyResults(pool(), task);
+
+  auto planStats = toPlanStats(task->taskStats());
+  auto& aggStats = planStats[aggId];
+  EXPECT_LT(aggStats.peakMemoryBytes, 204800);
```

Comment:
Did this test fail before the changes? What was peak memory usage before the fix? 

Let's also add tests for min_by and for inputs of type Varchar.

Reply:
Yeah, this check fails without the fix. The actual peak memory usage before the fix was 2428416 and after the fix is 134656. I've updated the unit test to check that the peak memory of min_by is the same as that of max_by and the peak memory is below 140000.